src/omni_epd/displays/inky_display.py

```diff
@@ -61,95 +54,62 @@ def __init__(self, deviceName, config):
-        self._device = deviceObj.Inky()
-
-        # set the width and height
-        self.width = self._device.width
-        self.height = self._device.height
+    def load_device(self, deviceName):
+        # need to figure out what type of device we have
+        dType, dColor, *_ = deviceName.split('_') + [None]
 
-        # get colors from the inky lib (won't be used normally as inky does conversion)
-        self.palette_filter = deviceObj.DESATURATED_PALETTE
+        if(dType == 'phat'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'phat')
+            device = deviceObj.InkyPHAT(dColor)
+        elif(dType == 'phat1608'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'phat')
+            device = deviceObj.InkyPHAT_SSD1608(dColor)
+        elif(dType == 'what'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'what')
+            device = deviceObj.InkyWHAT(dColor)
+        elif(dType == 'impression'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'inky_uc8159')
+            device = deviceObj.Inky()
+        elif(dType == 'auto'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'auto')
+            device = deviceObj.auto()
+
+        if(device.colour == 'multi'):
+            return device, device.CLEAN, 'color'
+        else:
+            return device, device.WHITE, dColor
 
     @staticmethod
     def get_supported_devices():
-        result = []
-
-        # python libs for this might not be installed - that's ok, return nothing
-        if(check_module_installed(INKY_PKG)):
-            # if passed return list of devices
-            result = [f"{INKY_PKG}.impression"]
-
-        return result
+        return [] if not check_module_installed(INKY_PKG) else [f"{INKY_PKG}.{n}" for n in InkyDisplay.deviceList]
 
+    # set the image and display
     def _display(self, image):
+        # set border
+        self._device.set_border(getattr(self._device, self._get_device_option('border', '').upper(), self._device.border_colour))
 
-        # no palette adjustments when color as the Inky lib does them from the image
-        if(self.mode == 'bw'):
+        # apply any needed conversions to this image based on the mode
+        if(self.mode == 'color'):
+            saturation = self._getfloat_device_option('saturation', .5)  # .5 is default from Inky lib
+            self._device.set_image(image.convert("RGB"), saturation=saturation)
+        else:
             image = self._filterImage(image)
+            self._device.set_image(image.convert("RGB"))
 
-        self._device.set_border(getattr(self._device, self._get_device_option('border', '').upper(), self._device.border_colour))
-        self._device.set_image(image.convert("RGB"), saturation=self._getfloat_device_option('saturation', .5))  # .5 is default from Inky lib
         self._device.show()
 
     def clear(self):
         for _ in range(2):

```

Comment:
does anyone know why this is done twice? 

removing `for _ in range(2):` makes no difference on either of my inky impressions

Reply:
You mean when clearing? (I think you selected the wrong lines) That's [how pimoroni does it](https://github.com/pimoroni/inky/blob/master/examples/7color/clear.py) on the impression, because it can leave ghosting behind. It doesn't necessarily need to be done twice, and doesn't need to at all on other devices AFAIK.

It might also be faster to set an image filled with `clear_color` instead of iterating through all the pixels.
`clear_image = Image.new("P", (self.width, self.height), self.clear_color)`

Reply:
[`device.set_image()`](https://github.com/pimoroni/inky/search?q=%22def+set_image%22) always writes to a buffer afaict, nothing goes to the screen until you call `device.show()`. but maybe this is best left alone for now.

I got a black screen after using `Image.new("P", (self.width, self.height), self.clear_color)` as part of the clear func btw, seems like it should work given the above.

Reply:
I take it back, it does seem to result in a white screen now. should I add it to the PR?

Reply:
If it works right, yeah.

At 600x448, I see a difference of 0.3 seconds on a pi 4, and 5 seconds on a pi zero 1.